### PR TITLE
Users list

### PR DESCRIPTION
**Pull request by octavioamu**
##### Checklist

- [ ] Read and conforms to the [contributor guidelines](https://docs.gitcoin.co/mk_contributors/)
- [ ] Read and conforms to the [style guidelines](https://docs.gitcoin.co/mk_styleguide/)
- [ ] Linter status: 100% pass
- [ ] Added relevant tests
- [ ] Tests pass and code coverage has not decreased
- [ ] Changes don't break existing behavior
- [ ] Commit message follows [commit guidelines](https://docs.gitcoin.co/mk_contributors/#step-4-commit)
- [ ] [Code documentation](https://docs.gitcoin.co/mk_contributors/#docstrings)
- [ ] Completed: Code review by core team and any requested changes

##### Refers/Fixes

Refs: #4019

###### Contributor

- [ ] Read and followed the [Contributor Guidelines](https://docs.gitcoin.co/mk_contributors/)
- [ ] Tested all changes **locally**
- [ ] Verified existing functionality
- [ ] Ran `make test` and everything passed!

###### Reviewer

- [ ] Affirm contributor guidelines have been followed and requested changes made
- [ ] CI tests and linting pass
- [ ] No conflicts (migrations, files, etc)
- [ ] Regression tested against staging or local deployment

###### Funder

- [ ] Validated requested changes were made to specification
- [ ] Bounty payout released to the contributor

### CODE REVIEW

**Comment by owocki** [comment]
hey @PixelantDesign @danlipert i spoke with @octavioamu this morning.  we were thinking about releasing this this week so it can be used in the ethereal hackathon ( what exists now => http://bits.owocki.com/1cba8654c7ba/Image%202019-04-10%20at%202.23.46%20PM.png is basic search )

then we will build in the fuller features https://github.com/gitcoinco/web/issues/4019 from there

i think we 
1. dont list this anywhere and just use it internally
2. slap a 'beta' tag on it
3. make it so you can only use it if logged in and you're an admin
4. make the PR in a non-draft state

how does that sound?

**Comment by PixelantDesign** [comment]
Works for me if it's in a good enough state. Though, I have not tested.

File: app/dashboard/templates/dashboard/users.html [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    <div class="" id="gc-users-directory" v-cloak>
+
+      <div class="container-fluid">
+        <nav class="navbar navbar-expand-lg navbar-light bg-light">
+          <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#usersFilters" aria-controls="usersFilters" aria-expanded="false" aria-label="Toggle navigation">
+            <span class="navbar-toggler-icon"></span>
+          </button>
+
+          <div class="collapse navbar-collapse" id="usersFilters">
+            <!-- <div class="navbar-nav mr-4">
```
Comment (danlipert) [inline app/dashboard/templates/dashboard/users.html]: Whats up with this fat comment? :)

File: app/dashboard/tests/test_users_list.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    """Define tests for the user list."""
+
+    def setUp(self):
+        self.request = RequestFactory()
+        for i in range(20):
+            user = User.objects.create(password="{}".format(i),
+                                       username="user{}".format(i))
+            profile = Profile.objects.create(user=user, data={}, handle="{}".format(i))
+
+    def test_user_list(self):
```
Comment (danlipert) [inline app/dashboard/tests/test_users_list.py]: nice! :D

File: app/dashboard/views.py [hunk 3]
```diff
@@ -731,6 +731,50 @@ def onboard(request, flow):
...
+def users_fetch(request):
+    """Handle displaying users."""
+    q = request.GET.get('search','')
+    limit = int(request.GET.get('limit', 10))
+    page = int(request.GET.get('page', 1))
+    order_by = request.GET.get('order_by', '-created_on')
+    context = {}
+    user_list = Profile.objects.all().order_by(order_by).filter(handle__icontains=q).cache()
+
+    # all_notifs = Notification.objects.filter(to_user_id=request.user.id).order_by('-id')
```
Comment (danlipert) [inline app/dashboard/views.py]: leftover comment

File: app/dashboard/views.py [hunk 4]
```diff
@@ -731,6 +731,50 @@ def onboard(request, flow):
...
+        'meta_title': "",
+        'meta_description': ""
+    }
+    return TemplateResponse(request, 'dashboard/users.html', params)
+
+
+@require_GET
+def users_fetch(request):
+    """Handle displaying users."""
+    q = request.GET.get('search','')
```
Comment (danlipert) [inline app/dashboard/views.py]: add a space before the final string here like `q = request.GET.get('search', '')`

File: app/dashboard/views.py [hunk 5]
```diff
@@ -731,6 +731,50 @@ def onboard(request, flow):
...
+    order_by = request.GET.get('order_by', '-created_on')
+    context = {}
+    user_list = Profile.objects.all().order_by(order_by).filter(handle__icontains=q).cache()
+
+    # all_notifs = Notification.objects.filter(to_user_id=request.user.id).order_by('-id')
+    params = dict()
+    all_pages = Paginator(user_list, limit)
+    all_users = []
+    for user in all_pages.page(page):
+        print(user)
```
Comment (danlipert) [inline app/dashboard/views.py]: leftover debugging print?

File: app/dashboard/views.py [hunk 6]
```diff
@@ -731,6 +731,50 @@ def onboard(request, flow):
...
+        profile_json = {}
+        profile_json = user.to_standard_dict()
+        if user.avatar_baseavatar_related.exists():
+            profile_json['avatar_id'] = user.avatar_baseavatar_related.first().pk
```
Comment (danlipert) [inline app/dashboard/views.py]: Better here to assign a var to `user.avatar_baseavatar_related.first()` and then access the `.pk` and `.avatar_url` from that var. Calling `.first()` multiple times re-queries the database

File: app/dashboard/templates/dashboard/users.html (hunk 1 above)
Comment (octavioamu) [reply app/dashboard/templates/dashboard/users.html]: is for the next version

**Pull request merged**